google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -124,8 +129,10 @@ private Publisher(Builder builder) throws IOException {
 
     // Publisher used to take maxAttempt == 0 to mean infinity, but to GAX it means don't retry.
     // We post-process this here to keep backward-compatibility.
+    // Also, if "message ordering" is enabled, the publisher should retry sending the failed
+    // message infinitely rather than sending the next one.
     RetrySettings retrySettings = builder.retrySettings;
-    if (retrySettings.getMaxAttempts() == 0) {
+    if (retrySettings.getMaxAttempts() == 0 || builder.enableMessageOrdering) {
```

Comment:
Here we're using builder.enableMessageOrdering and on line 203 we're using enableMessageOrdering. I would guess this one should change to enableMessageOrdering?